src/ui/field/Field.tsx

```diff
@@ -2,11 +2,42 @@ import React from 'react';
 
 import ErrorIcon from '@material-ui/icons/Error';
 
-import { FieldBaseProps, FieldProps, TextareaFieldProps } from '.';
-
 import csx from './Field.scss';
 
-export const FieldBase = ({ label, error, children, className = '' }: FieldBaseProps) => {
+// TODO
+// Move components to separate files with their namespace definitions and put them at ui/name-of-component
+
+export namespace FieldBase {
+  export interface Props {
+    label: string;
+    children: React.ReactNode;
+    className?: string;
+    error?: string;
+  }
+}
+
+export namespace Field {
+  export interface Props
+    extends React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement> {
+    label: string;
+    className?: string;
+    error?: string;
+  }
+}
+
+export namespace TextareaField {
+  export interface Props
+    extends React.DetailedHTMLProps<
+      React.TextareaHTMLAttributes<HTMLTextAreaElement>,
+      HTMLTextAreaElement
+    > {
+    label: string;
+    className?: string;
+    error?: string;
+  }
```

Comment:
Why are we exporting those namespaces? I guess it shouldn't be exported?

Reply:
It should be. Because those components should be separated with files. I added them initially in single file and later they should be moved to separate files.